sw/applications/example_spi_flash_write_quad/main.c

```diff
@@ -0,0 +1,446 @@
+            .csaat      = true,
+            .speed      = kSpiSpeedStandard,
+            .direction  = kSpiDirTxOnly
+        });
+        uint32_t spi_status_read_cmd = spi_create_command((spi_command_t){
+            .len        = 0,
+            .csaat      = false,
+            .speed      = kSpiSpeedStandard,
+            .direction  = kSpiDirRxOnly
+        });
+        spi_set_command(&spi_host, spi_status_cmd);
+        spi_wait_for_ready(&spi_host);
+        spi_set_command(&spi_host, spi_status_read_cmd);
+        spi_wait_for_ready(&spi_host);
+        spi_wait_for_rx_watermark(&spi_host);
+        spi_read_word(&spi_host, &flash_resp[0]);
+        if ((flash_resp[0] & 0x01) == 0) flash_busy = false;
+    }
+}
+
+
+int main(int argc, char *argv[]) {
+    soc_ctrl_t soc_ctrl;
+    soc_ctrl.base_addr = mmio_region_from_addr((uintptr_t)SOC_CTRL_START_ADDRESS);
+
+    #ifdef USE_SPI_FLASH
+    if ( get_spi_flash_mode(&soc_ctrl) == SOC_CTRL_SPI_FLASH_MODE_SPIMEMIO )
+    {
+        PRINTF("This application cannot work with the memory mapped SPI FLASH module - do not use the FLASH_EXEC linker script for this application\n");
+        return EXIT_SUCCESS;
+    }
+    #endif
+
+    /*
+     * Configure the SPI<->flash connection parameters and the SPI host.
+     * Set the write enale bit.
+    */
+    spi_config();
+
+    /*
+     * Set the QE bit in the W25Q128JW flash memory.
+     * The function execute the command only if the target is the PYNQ-Z2 FPGA.
+    */
+    set_QE_bit();
+
+    // Write enable
+    const uint32_t write_enable_cmd = 0x06;
+    spi_write_word(&spi_host, write_enable_cmd);
+    const uint32_t cmd_write_en = spi_create_command((spi_command_t){
+        .len        = 0,
+        .csaat      = false,
+        .speed      = kSpiSpeedStandard,
+        .direction  = kSpiDirTxOnly
+    });
+    spi_set_command(&spi_host, cmd_write_en);
+    spi_wait_for_ready(&spi_host);
+
+    // Write command - Program Page + address
+    const uint32_t write_byte_cmd = ((REVERT_24b_ADDR(FLASH_ADDR) << 8) | 0x32); // QUAD SPEED COMMAND
+    spi_write_word(&spi_host, write_byte_cmd);
+    const uint32_t cmd_write = spi_create_command((spi_command_t){
+        .len        = 3,
+        .csaat      = true, 
+        .speed      = kSpiSpeedStandard,
+        .direction  = kSpiDirTxOnly
+    });
+    spi_set_command(&spi_host, cmd_write);
+    spi_wait_for_ready(&spi_host);
+
+    // Write test buffer to SPI TX FIFO
+    for (int i = 0; i < 16; i++) {
+        spi_write_word(&spi_host, flash_data_towrite[i]);
+    }
+    const uint32_t cmd_write1 = spi_create_command((spi_command_t){
+        .len        = 16*4 - 1,
+        .csaat      = false, 
+        .speed      = kSpiSpeedQuad, // QUAD SPEED 
+        .direction  = kSpiDirTxOnly
+    });
+    spi_set_command(&spi_host, cmd_write1);
+    spi_wait_for_ready(&spi_host);
+
+    // Wait for all the fifo to be drained
+    spi_wait_for_tx_empty(&spi_host);
+    #ifndef TARGET_SIM
```

Comment:
let's wait for response also in simulation please

Reply:
It will only work in questa, right?

Reply:
Same problem, the function will never return.